pkg/openstack/loadbalancer.go

```diff
@@ -605,6 +606,32 @@ func (lbaas *LbaasV2) getLoadBalancerLegacyName(_ context.Context, _ string, ser
 	return cloudprovider.DefaultLoadBalancerName(service)
 }
 
+// this function gets the listeners tags from the service annotation and   
+func (lbaas *LbaasV2) customLoadBalanceListenerTag(service *corev1.Service) ([]string, error) {
+	var listener listeners.Listener
+	annotationVal := getStringFromServiceAnnotation(service, ServiceAnnotationLoadBalancerListenerTags, "")
+
+	listOfTags := strings.Split(annotationVal, "=")
+
+	if annotationVal == "" {
+		return listener.Tags, fmt.Errorf("annotation is missing or empty for %q", service.Name)
+	}
+
+	if annotationVal != "" {
+		for _, tags := range listOfTags {
+			tag := strings.Split(strings.TrimSpace(tags), "=")
+
+			if len(tag) == 2 {
+				listener.Tags = append(listener.Tags, tags)
+			} else {
+				return listener.Tags, fmt.Errorf("invalid tag format: %s", tags)
+			}
```

Comment:
This code doesn't allow for a tag to contain `=` in the value. Is that expected? Is this restriction necessary?

If not, and a `=` is an acceptable character in this context, then consider using [`strings.Cut`](https://pkg.go.dev/strings#Cut) instead (see [this example](https://go.dev/play/p/DG3gj7BShcE)).